neighsyncd/neighsync.cpp

```diff
@@ -11,12 +11,244 @@
 #include "linkcache.h"
 
 #include "neighsync.h"
+#include "warm_restart.h"
 
 using namespace std;
 using namespace swss;
 
-NeighSync::NeighSync(DBConnector *db) :
-    m_neighTable(db, APP_NEIGH_TABLE_NAME)
+const NeighRestartAssit::cache_state_map NeighRestartAssit::cacheStateMap =
+{
+    {STALE,     "STALE"},
+    {SAME,      "SAME"},
+    {NEW,       "NEW"},
+    {DELETE,    "DELETE"},
+    {UNKNOWN,   "UNKNOWN"}
+};
+
+NeighRestartAssit::NeighRestartAssit(RedisPipeline *pipelineAppDB,
+  RedisPipeline *pipelineConfDB):
+    m_appTable(pipelineAppDB, APP_NEIGH_TABLE_NAME, false),
+    m_cfgTable(pipelineConfDB, CFG_WARM_RESTART_TABLE_NAME, false)
+{
+
+    m_appTableName = m_appTable.getTableName();
+
+    if (!WarmStart::isWarmStart())
+    {
+        m_warmStartInProgress = false;
+    }
+    else
+    {
+        uint32_t temp_value = WarmStart::getWarmStartTimer("neighsyncd");
+        if (temp_value != 0)
+        {
+            m_reconcileTimer = temp_value;
+        }
+    }
+}
+
+NeighRestartAssit::~NeighRestartAssit()
+{
+}
+
+/* join the field-value strings for straight printing */
+string NeighRestartAssit::joinVectorString(const vector<FieldValueTuple> &fv)
+{
+    string s;
+    for (const auto &temps : fv )
+    {
+	   s += temps.first + ":" + temps.second + ", ";
+    }
+    return s;
+}
+
+void NeighRestartAssit::setCacheEntryState(std::vector<FieldValueTuple> &fvVector,
```

Comment:
>setCacheEntryState [](start = 24, length = 18)

setCacheLastEntryState

Reply:
The state was meant for the entry not the last,  just that we keep the state at the tail of the entry.